services/checks/registry.go

```diff
@@ -91,24 +90,19 @@ func (r *Registry) CreateAlert(id string, labels, annotations map[string]string,
 	}
 }
 
-// RemovePrefix removes all alerts with given ID prefix except a given list of IDs.
-func (r *Registry) RemovePrefix(prefix string, keepIDs map[string]struct{}) {
+// removePrefix removes all alerts with given ID prefix except a given list of IDs.
+func (r *registry) clean() {
 	r.rw.Lock()
 	defer r.rw.Unlock()
 
 	for id := range r.alerts {
-		if _, ok := keepIDs[id]; ok {
-			continue
-		}
-		if strings.HasPrefix(id, prefix) {
-			delete(r.alerts, id)
-			delete(r.times, id)
-		}
+		delete(r.alerts, id)
+		delete(r.times, id)
 	}
```

Comment:
It can be simply
```
r.alerts = make(map[string]ammodels.PostableAlert)
r.times = make(map[string]time.Time)
```

Reply:
Funny enough, for/range/delete loop is optimised by the compiler: https://golang.org/doc/go1.11#performance-compiler
(I do not imply that we need that optimisation there, that's JFYI)